controllers/configuration_controller.go

```diff
@@ -337,112 +276,148 @@ func (r *ConfigurationReconciler) reconcile(ctx context.Context, cv *v1alpha1.Co
+		}
 
-	obj, ok := source.(v1beta2.Source)
-	if !ok {
-		return "", fmt.Errorf("not a source: %v", obj)
+		return makeRequestsForConfigurations(append(sourceRefs.Items, configRefs.Items...)...)
 	}
-
-	return obj.GetArtifact().Checksum, nil
 }
 
-func (r *ConfigurationReconciler) requestsForRevisionChangeOf(indexKey string) func(obj client.Object) []reconcile.Request {
+// this function will enqueue a reconciliation for any component version which is referenced
+// in the .spec.sourceRef or spec.configRef field of a Configuration
+func (r *ConfigurationReconciler) findObjectsForGitRepository(key string) func(client.Object) []reconcile.Request {
 	return func(obj client.Object) []reconcile.Request {
-		snap, ok := obj.(*v1alpha1.Snapshot)
-		if !ok {
-			panic(fmt.Sprintf("expected snapshot but got: %T", obj))
+		patchRefs := &v1alpha1.ConfigurationList{}
+		if err := r.List(context.TODO(), patchRefs, &client.ListOptions{
+			FieldSelector: fields.OneTermEqualSelector(key, obj.GetName()),
+			Namespace:     obj.GetNamespace(),
+		}); err != nil {
+			return []reconcile.Request{}
 		}
 
-		if snap.Status.LastReconciledDigest == "" {
-			return nil
-		}
+		return makeRequestsForConfigurations(patchRefs.Items...)
+	}
+}
 
-		ctx := context.Background()
-		var list v1alpha1.ConfigurationList
-		if err := r.List(ctx, &list, client.MatchingFields{
-			indexKey: client.ObjectKeyFromObject(obj).String(),
-		}); err != nil {
-			return nil
+func (r *ConfigurationReconciler) checkReadiness(ctx context.Context, ns string, obj *v1alpha1.ObjectReference) bool {
+	var ref conditions.Getter
+	switch obj.Kind {
+	case v1alpha1.ComponentVersionKind:
+		if obj.Namespace == "" {
+			obj.Namespace = ns
 		}
-
-		var reqs []reconcile.Request
-		for _, d := range list.Items {
-			if snap.Status.LastReconciledDigest == d.Status.LatestSnapshotDigest {
-				continue
-			}
-			reqs = append(reqs, reconcile.Request{
-				NamespacedName: types.NamespacedName{
-					Namespace: d.Namespace,
-					Name:      d.Name,
-				},
-			})
+		ref = &v1alpha1.ComponentVersion{}
+		if err := r.Get(ctx, obj.GetObjectKey(), ref); err != nil {
+			return false
 		}
 
-		return reqs
+	default:
+		// if the APIVersion is not set then default to "delivery.ocm.software/v1alpha1"
+		if obj.APIVersion == "" {
+			obj.APIVersion = v1alpha1.GroupVersion.String()
+		}
+		// if the Namespace is not set then default to the parent object's namespace
+		if obj.Namespace == "" {
+			obj.Namespace = ns
+		}
+		// the dynamic client needs to know the GroupVersionResource for the object it's trying to fetch
+		// so construct that and fetch the unstructured object
+		gvr := obj.GetGVR()
+		src, err := r.DynamicClient.Resource(gvr).Namespace(obj.Namespace).Get(ctx, obj.Name, metav1.GetOptions{})
+		if err != nil {
+			return false
+		}
+		snapshotName, ok, err := unstructured.NestedString(src.Object, "status", "snapshotName")
+		if err != nil {
+			return false
+		}
+		if !ok {
+			return false
+		}
```

Comment:
Throw in some debug lines into these false's with errors. I just _know_ it will fail for some stupid reason and we won't know why because it just returns false on an error.. :D